codalab/rest/bundles.py

```diff
@@ -747,7 +753,12 @@ def _update_bundle_contents_blob(uuid):
         if request.query.filename:
             filename = request.query.get('filename', default='contents')
             sources = [(filename, request['wsgi.input'])]
-        if sources:
+        bundle_link_url = getattr(bundle.metadata, "link_url", None)
```

Comment:
why do we move this check into this function?

Reply:
It seems to semantically make more sense -- why would we call `upload_to_bundle_store` if we just end up not doing anything in that function linked bundles? Better to just not call `upload_to_bundle_store` in the first place.

Later, I also add more complexity to `upload_to_bundle_store` (#3308), so that's why I go ahead and move this check out of that function in this PR.

Reply:
Makes sense, got a bit lost on the hierarchy on functions lol